### ISSUE TITLE
mode of  an intermediate layer's output

### ISSUE THREAD

**Issue by cchhdd** [MoyanZitto/keras-cn#23]
`get_3rd_layer_output = K.function([model.layers[0].input, K.learning_phase()],
                                  [model.layers[3].output])
# output in test mode = 0

layer_output = get_3rd_layer_output([X, 0])[0]
# output in train mode = 1

layer_output = get_3rd_layer_output([X, 1])[0]`

I've found that the result of two modes are different.
what is the different between test mode and train mode?
what does these two modes mean?

thank u

**Comment by MoyanZitto** [comment]
you can use 

`````` python
    ```python
    your code
    ```
``````

to make your code clear.

Yes, definitely the two resutls are different, the flag "train_mode" indicate the mode of the model. Some layers, BN and Dropout for example, have different behaviors in different mode.

You may check corresponding papers for a deeper understanding.

**Comment by cchhdd** [comment]
@MoyanZitto thanks a lot for your help .I've understood what u mean
